server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/InternalVariableWidthHistogram.java

```diff
@@ -471,6 +471,9 @@ private void mergeBucketsWithSameMin(List<Bucket> buckets, ReduceContext reduceC
             BucketRange nextRange = ranges.get(i+1);
             if(range.min == nextRange.min){
                 range.mergeWith(nextRange);
+                ranges.remove(i+1);
+            } else{
+                i++;
```

Comment:
If you are going to manipulate `i` outside of the `for` line, would you kindly replace this with a `while` loop? When I see `while` I'm automatically on guard for all kinds of sneaky stuff. But `for` lulls me into a false sense of security.